app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+            format: {with: VALID_EMAIL_REGEX},
+            uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
has_secure_password là gì nhỉ, nó dùng để làm gì vậy

Reply:
has_secure_password chỉ ra rằng user sẽ có một password, password này khi người dùng nhập vào sẽ có 2 chỗ : password và password_confirmation, 2 pass này phải giống nhau. Sau đó pass đó được đưa vào bảng băm và lưu vào database dưới dạng password_digest
Tác dụng : bảo mật
password_digest : password đã được đi qua bảng băm, không thể đọc được 
password and password_confirmation : bao gồm password và password lặp lại phải nhập 1 lần nữa
authenticate :trả về user khi pass đúng và fasle khi sai